### PR TITLE
pitr: prevent from restore point to cluster running log backup

### PR DESCRIPTION
**Pull request by Leavrth**
Signed-off-by: Leavrth <jianjun.liao@outlook.com>

### What problem does this PR solve?

Issue Number: ref #40797 

Problem Summary:
If restore point to the cluster running log backup task, the backup files will be contaminated.
### What is changed and how it works?
Before run resotre point, check whether there is a log backup task running in the cluster.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -1034,6 +1034,28 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
...
+		return err
+	}
+	cli := streamhelper.NewMetaDataClient(etcdCLI)
+	defer cli.Close()
+	tasks, err := cli.GetAllTasks(ctx)
+	if err != nil {
+		return err
+	}
+	if len(tasks) > 0 {
+		return errors.Errorf("There is a log backup task running: %s, please stop the task before restore", tasks[0].Info.Name)
```
Comment (joccau) [inline br/pkg/task/stream.go:1054]:
```suggestion
		return errors.Errorf("log backup task is running: %s, please stop the task before restore, and after PITR operation finished, create log-backup task again and create a full backup on this cluster.", tasks[0].Info.Name)
```

**Review by joccau — APPROVED** [review]
rest LGTM, need add an integration test.

File: br/pkg/restore/client.go [hunk 2]
```diff
@@ -2101,15 +2101,16 @@ func (rc *Client) RestoreKVFiles(
 		return errors.Trace(err)
 	})
 
+	if err = eg.Wait(); err != nil {
```
Comment (3pointer) [inline br/pkg/restore/client.go:2104]: why move this here?

**Review by 3pointer — APPROVED** [review]

File: br/pkg/restore/client.go (hunk 2 above)
Comment (Leavrth) [reply br/pkg/restore/client.go:2104]: The log info below need `skipFile`, which is updated by these worker functions. So we need to wait these functions to finish at first, and then print the log Info.

**Pull request merged**